File: data/derived_cot/rq3_prompting/results/translation/r1/cpp_to_py/code_output/DataStatistics.py
class DataStatistics:
    @staticmethod
    def _round_half_up(x: float) -> int:
        """Round to the nearest integer, ties away from zero (like C++ std::round)."""
        if x >= 0:
            return int(x + 0.5)
        else:
            return int(x - 0.5)

    def mean(self, data: list[float]) -> float:
        if not data:
            raise ValueError("Data vector is empty.")
        total = sum(data)
        # round to two decimal places
        return self._round_half_up(total / len(data) * 100) / 100.0

    def median(self, data: list[float]) -> float:
        if not data:
            raise ValueError("Data vector is empty.")
        sorted_data = sorted(data)
        n = len(sorted_data)
        if n % 2 == 0:
            # even: average of two middle values, rounded
            avg = (sorted_data[n // 2 - 1] + sorted_data[n // 2]) / 2.0
            return self._round_half_up(avg * 100) / 100.0
        else:
            # odd: middle value directly (no rounding)
            return sorted_data[n // 2]

    def mode(self, data: list[int]) -> list[int]:
        if not data:
            raise ValueError("Data vector is empty.")
        # count frequencies
        count_map = {}
        for num in data:
            count_map[num] = count_map.get(num, 0) + 1
        # find maximum frequency
        max_count = max(count_map.values())
        # collect all numbers with that frequency
        modes = [num for num, cnt in count_map.items() if cnt == max_count]
        # ensure sorted order (matches C++ map iteration)
        modes.sort()
        return modes

File: data/derived_cot/rq3_prompting/results/translation/qwen/java_to_cpp/code_output/DataStatistics.cpp
#include <vector>
#include <algorithm>
#include <cmath>
#include <unordered_map>

class DataStatistics {
public:
    double mean(const std::vector<int>& data) {
        if (data.empty()) {
            throw std::domain_error("DataStatistics::mean: empty array");
        }
        double sum = 0.0;
        for (int num : data) {
            sum += num;
        }
        double average = sum / data.size();
        long rounded = static_cast<long>(std::round(average * 100.0));
        return static_cast<double>(rounded) / 100.0;
    }

    double median(std::vector<int> data) {
        if (data.empty()) {
            throw std::domain_error("DataStatistics::median: empty array");
        }
        std::sort(data.begin(), data.end());
        int n = data.size();

        if (n % 2 == 0) {
            int middle = n / 2;
            double average = (static_cast<double>(data[middle - 1]) + data[middle]) / 2.0;
            long rounded = static_cast<long>(std::round(average * 100.0));
            return static_cast<double>(rounded) / 100.0;
        } else {
            int middle = n / 2;
            return data[middle];
        }
    }

    std::vector<int> mode(std::vector<int> data) {
        if (data.empty()) {
            return {};
        }

        std::unordered_map<int, long> frequencyMap;
        for (int num : data) {
            frequencyMap[num]++;
        }

        long maxCount = 0;
        for (const auto& pair : frequencyMap) {
            if (pair.second > maxCount) {
                maxCount = pair.second;
            }
        }

        std::vector<int> modes;
        for (const auto& pair : frequencyMap) {
            if (pair.second == maxCount) {
                modes.push_back(pair.first);
            }
        }

        std::sort(modes.begin(), modes.end());
        return modes;
    }
};